plugin/src/test/scala/io/github/retronym/classpathshrinker/ClassPathShrinkerWithTargetsSpec.scala

```diff
@@ -0,0 +1,107 @@
+package io.github.retronym.classpathshrinker
+
+import coursier.{Dependency, Module}
+import io.github.retronym.classpathshrinker.TestUtil._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(classOf[JUnit4])
+class ClassPathShrinkerWithTargetsSpec {
+
+  object Dependencies {
+    val commons =
+      Dependency(Module("org.apache.commons", "commons-lang3"), "3.5")
+    val guava = Dependency(Module("com.google.guava", "guava"), "21.0")
+  }
+
+  import Dependencies._
+
+  @Test
+  def `unused jars are not reported when targets are used`(): Unit = {
+    val testCode =
+      """object Demo"""
+    val commonsPath = Coursier.getArtifact(commons)
+    val commonsTarget = "commonsTarget"
+    val indirect = Map(commonsPath -> commonsTarget)
+    run(testCode, withIndirect = indirect).expectNoJarMessageOn(commonsPath)
+  }
+
+
+  @Test
+  def `error on indirect dependency target`(): Unit = {
+    val testCode =
+      """object Foo {
+        |  org.apache.commons.lang3.ArrayUtils.EMPTY_BOOLEAN_ARRAY.length
+        |}
+      """.stripMargin
+    val commonsPath = Coursier.getArtifact(commons)
+    val commonsTarget = "//commons:Target".encode()
+    val indirect = Map(commonsPath -> commonsTarget)
+    run(testCode, withIndirect = indirect).expectErrorOn(indirect(commonsPath).decoded)
+  }
+
+  @Test
+  def `error on multiple indirect dependency targets`(): Unit = {
+    val testCode =
+      """object Foo {
+        |  org.apache.commons.lang3.ArrayUtils.EMPTY_BOOLEAN_ARRAY.length
+        |  com.google.common.base.Strings.commonPrefix("abc", "abcd")
+        |}
+      """.stripMargin
+    val commonsPath = Coursier.getArtifact(commons)
+    val commonsTarget = "commonsTarget"
+
+    val guavaPath = Coursier.getArtifact(guava)
+    val guavaTarget = "guavaTarget"
+
+    val indirect = Map(commonsPath -> commonsTarget, guavaPath -> guavaTarget)
+    run(testCode, withIndirect = indirect).expectErrorOn(commonsTarget, guavaTarget)
+  }
+
+  @Test
+  def `do not give error on direct dependency target`(): Unit = {
```

Comment:
Do we need another test with multiple direct dependencies? Maybe just change this one to multiple?